main.nf

```diff
@@ -301,6 +303,7 @@ if (params.skip_basecalling) {
         barcode_kit = params.barcode_kit ? "--barcode_kits $params.barcode_kit" : ""
         config = params.guppy_config ? "--config $params.guppy_config" : "--flowcell $params.flowcell --kit $params.kit"
         proc_options = params.guppy_gpu ? "--device $params.gpu_device --num_callers $task.cpus --cpu_threads_per_caller $params.guppy_cpu_threads --gpu_runners_per_device $params.guppy_gpu_runners" : "--num_callers 2 --cpu_threads_per_caller ${task.cpus/2}"
+        model = params.guppy_model ? "--model $params.guppy_model" : ""
```

Comment:
We wont be able to pass the file to the process like this @lwratten . It will have to be staged properly in a channel by NF and then passed optionally if it exists. I had similar issues with the `blacklist` files I was using for the `chipseq` pipeline.

The only solution I found to overcome this was by staging the file first like so:
https://github.com/nf-core/chipseq/blob/21be3149542cdc84431e12d1e092359058aed32a/main.nf#L166

And then providing logic to check if the channel is empty or not within the process:
https://github.com/nf-core/chipseq/blob/21be3149542cdc84431e12d1e092359058aed32a/main.nf#L424

Reply:
The problem I'm having is that the "file" can be a local file i.e. a custom basecalling model, or it can be the name of the model which is stored in `/opt/ont/guppy` - there are at least 2 inbuilt ones there. 

This is the same issue as the `--config` file too - it can be a preset in `/opt/ont/guppy` or a local file.

If the path is inputted as a string, it seems to pass correctly, but when staged as a file the path is always removed and Guppy searches in `/opt/ont/guppy`, can't find the file and exits.

In the below example I ran the pipeline with a custom `config` and `model` file - the `config` path is passed correctly and located, while the `model` fails when staged as a file
![image](https://user-images.githubusercontent.com/37123979/69117948-7b82d780-0acc-11ea-9937-b2db37a394f3.png)

Is there a reason we can't just pass it as a string and must stage it as a file in this instance when it is working for `--config`?

Reply:
Yes, I suspect `--config` needs to be changed too. Not ideal that the same parameter can be a string or a file...maybe we will have to have a `try...except` block in the pipeline to check whether the input is a string or a file before staging it using "file" or "fromFilePath".

See: https://github.com/nf-core/atacseq/issues/46#issuecomment-535626037